packages/cactus-plugin-cc-tx-visualization/src/main/typescript/plugin-cc-tx-visualization.ts

```diff
@@ -0,0 +1,580 @@
+          ),
+          latestUpdate: lastAggregated,
+          revenue: revenue,
+        };
+        this.crossChainModel.setCCTxs(key, metrics);
+      }
+    });
+    this.crossChainModel.setLastAggregationDate(newAggregationDate);
+    const finalTime = new Date();
+    this.log.debug(
+      `${this.className}-AGGREGATE-CCTX-SUCCESS:${finalTime.getTime() - startTime.getTime()}`,
+    );
+    return;
+  }
+
+  public async persistCrossChainLogCsv(name?: string): Promise<string> {
+    const startTime = new Date();
+    const columns = this.crossChainLog.getCrossChainLogAttributes();
+    const logName = name
+      ? `${name}.csv`
+      : `cctxviz_log_${startTime.getTime()}.csv`;
+    const csvFolder = path.join(__dirname, "../", "../", "test", "csv");
+    const logPath = path.join(csvFolder, logName);
+    const fnTag = `${this.className}#persistCrossChainLogCsv()`;
+
+    stringify(
+      this.crossChainLog.logEntries,
+      {
+        header: true,
+        columns: columns,
+        delimiter: ";",
+      },
+      (err: Error | undefined, data: string) => {
+        if (err) {
+          const errorMessage = `${fnTag} Failed to stringify log: ${err}`;
+          throw new RuntimeError(errorMessage, err);
+        }
+        this.log.debug(data);
+
+        // Create directory if it doesn't exist
+        if (!fs.existsSync(csvFolder)) {
+          fs.mkdirSync(csvFolder);
+        }
+        fs.writeFileSync(logPath, data);
+      },
+    );
+
+    const finalTime = new Date();
+    this.log.debug(
+      `EVAL-${this.className}-PERSIST-LOG-CVS:${finalTime.getTime() - startTime.getTime()}`,
+    );
+    return logName;
+  }
+
+  public async persistCrossChainLogJson(name?: string): Promise<string> {
+    const startTime = new Date();
+    const logName = name
+      ? `${name}.json`
+      : `cctxviz_log_${startTime.getTime()}.json`;
+    const jsonFolder = path.join(__dirname, "../", "../", "test", "json");
+    const logPath = path.join(jsonFolder, logName);
+    const fnTag = `${this.className}#persistCrossChainLogJson()`;
+
+    try {
+      const data = JSON.stringify(this.crossChainLog.logEntries, null, 2);
+      this.log.debug(data);
+
+      // Create directory if it doesn't exist
+      if (!fs.existsSync(jsonFolder)) {
+        fs.mkdirSync(jsonFolder);
+      }
+      fs.writeFileSync(logPath, data);
+
+      const finalTime = new Date();
+      this.log.debug(
+        `EVAL-${this.className}-PERSIST-LOG-JSON:${finalTime.getTime() - startTime.getTime()}`,
+      );
+
+      return logName;
+    } catch (error) {
+      const errorMessage = `${fnTag} Failed to stringify log: ${error}`;
+      throw new RuntimeError(errorMessage, error);
+    }
+  }
+
+  // Receives a serialized model
+  public async saveModel(
+    modelType: CrossChainModelType,
+    model: string,
+  ): Promise<void> {
+    this.crossChainModel.saveModel(modelType, model);
+  }
```

Comment:
we should also have methods to:
1) read a serialized CrossChainModelType
2) read the output from pm4py
3) convert pm4py into CrossChainModelType, and vice versa

Reply:
I recommend you to create a new directory, something like `pm4py-adapter`, where you put the code that handles interactions between this plugin and pm4py